shell/browser/ui/cocoa/electron_ns_window_delegate.mm

```diff
@@ -128,27 +128,62 @@ - (void)windowDidResignKey:(NSNotification*)notification {
 - (NSSize)windowWillResize:(NSWindow*)sender toSize:(NSSize)frameSize {
   NSSize newSize = frameSize;
   double aspectRatio = shell_->GetAspectRatio();
+  NSWindow* window = shell_->GetNativeWindow().GetNativeNSWindow();
 
   if (aspectRatio > 0.0) {
     gfx::Size windowSize = shell_->GetSize();
     gfx::Size contentSize = shell_->GetContentSize();
     gfx::Size extraSize = shell_->GetAspectRatioExtraSize();
 
+    double titleBarHeight = windowSize.height() - contentSize.height();
     double extraWidthPlusFrame =
         windowSize.width() - contentSize.width() + extraSize.width();
-    double extraHeightPlusFrame =
-        windowSize.height() - contentSize.height() + extraSize.height();
+    double extraHeightPlusFrame = titleBarHeight + extraSize.height();
 
     newSize.width =
         roundf((frameSize.height - extraHeightPlusFrame) * aspectRatio +
                extraWidthPlusFrame);
     newSize.height =
         roundf((newSize.width - extraWidthPlusFrame) / aspectRatio +
                extraHeightPlusFrame);
+
+    // Clamp to minimum width/height while ensuring aspect ratio remains.
+    NSSize minSize = [window minSize];
+    NSSize zeroSize =
+        shell_->has_frame() ? NSMakeSize(0, titleBarHeight) : NSZeroSize;
+    if (!NSEqualSizes(minSize, zeroSize)) {
+      double minWidthForAspectRatio =
+          (minSize.height - titleBarHeight) * aspectRatio;
+      bool atMinHeight =
+          minSize.height > zeroSize.height && newSize.height <= minSize.height;
+      newSize.width = atMinHeight ? minWidthForAspectRatio
+                                  : std::max(newSize.width, minSize.width);
+
+      double minHeightForAspectRatio = minSize.width / aspectRatio;
```

Comment:
I don't know how verbose comments in electron are, but I had to read this and actually derive these values to believe them. Might be helpful to have the formula:
 
If aspect ratio is r and `r = w/h`
Then `w = r*h`
and `h = w/r`

Reply:
> actually derive these values to believe them

Can you explain what you mean here? What kind of comment would you expect to see?

Reply:
I'm an excessive commenter so take this with a grain of salt - I know all devs and all projects have different norms and commenting debates can get spicy.

When I initially read the code I was surprised that we were using width to derive the height of the window. Of course, that's necessary when locking the aspect ratio. But I haven't calculated an aspect ratio in a while, so I looked up the formula and went from there. After that, it made sense.

If I were writing this code, I'd just note the formula for other people who forgot that `r = w/h`. But I think it's equally valid to put no comments.